watertap/tools/oli_api/client.py

```diff
@@ -321,122 +357,114 @@ def _delete_dbs_file(self, dbs_file_id):
-        if tee:
-            print(results_link)
-        # TODO: raise error
-        if results_link == "":
-            return dict()
-
-        # poll on results link until success
-        data = ""
-        endpoint = results_link
-        method = "GET"
-        request_iter = 0
-
-        while True:
-            # make request and time
-            start_time = time.time()
-            request_result2 = self.request_auto_login(add_additional_header)
-            end_time = time.time()
-            request_time = end_time - start_time
-            if tee:
-                print("Second request time =", request_time)
-
-            # check if max requests exceeded
-            request_iter = request_iter + 1
-            if request_iter > max_request:
-                break
-
-            # extract
-            if tee:
-                print(request_result2)
-            if bool(request_result2):
-                if "status" in request_result2:
-                    status = request_result2["status"]
-                    if tee:
-                        print(status)
-                    if status == "PROCESSED" or status == "FAILED":
-                        if "data" in request_result2:
-                            return request_result2["data"]
-                        else:
-                            break
-                    elif status == "IN QUEUE" or status == "IN PROGRESS":
-                        if poll_time > request_time:
-                            time.sleep(poll_time - request_time)
-                        continue
+                            _logger.info(
+                                f"{get_result['message']} after {round(time.time() - start_time,2)}s"
+                            )
+                            return get_result["data"]
                     else:
-                        break
+                        poll_timer = time.time() - last_poll_time
                 else:
-                    break
-            else:
-                break
-        # TODO: raise error
-        return dict()
+                    raise RuntimeError("Poll limit exceeded.")
+
+    def get_result_link(self, result):
+        """
+        Get result link from OLI Cloud POST request.
+
+        :param result: json containing result of POST request
+
+        :return result_link: string indicating URL to access call results
+        """
+        result_link = ""
+        if bool(result):
+            if "data" in result:
+                if "status" in result["data"]:
+                    if "resultsLink" in result["data"]:
+                        _logger.debug(
+                            f"Call status is {result['data']['status']},"
+                            + f" result link is {result['data']['resultsLink']}"
+                        )
+                        result_link = result["data"]["resultsLink"]
+        else:
+            _logger.debug("No result link found")
```

Comment:
So, this message will only be displayed in DEBUG mode, but an exception will always be raised nevertheless, correct (e.g., line 420)?